Bugfix: iscsi initiator check causes crash

isbm opened the pull request:
### What does this PR do?

Fixes this (affects Salt SSH and anywhere iSCSI initiator is configured):
```
    stderr:
        Traceback (most recent call last):
          File "/var/tmp/.root_26386f_salt/salt-call", line 15, in <module>
            salt_call()
          File "/var/tmp/.root_26386f_salt/py2/salt/scripts.py", line 395, in salt_call
            client.run()
          File "/var/tmp/.root_26386f_salt/py2/salt/cli/call.py", line 47, in run
            caller = salt.cli.caller.Caller.factory(self.config)
          File "/var/tmp/.root_26386f_salt/py2/salt/cli/caller.py", line 79, in factory
            return ZeroMQCaller(opts, **kwargs)
          File "/var/tmp/.root_26386f_salt/py2/salt/cli/caller.py", line 289, in __init__
            super(ZeroMQCaller, self).__init__(opts)
          File "/var/tmp/.root_26386f_salt/py2/salt/cli/caller.py", line 102, in __init__
            self.minion = salt.minion.SMinion(opts)
          File "/var/tmp/.root_26386f_salt/py2/salt/minion.py", line 716, in __init__
            opts[u'grains'] = salt.loader.grains(opts)
          File "/var/tmp/.root_26386f_salt/py2/salt/loader.py", line 729, in grains
            ret = funcs[key]()
          File "/var/tmp/.root_26386f_salt/py2/salt/grains/core.py", line 2459, in iscsi_iqn
            grains['iscsi_iqn'] = _linux_iqn()
          File "/var/tmp/.root_26386f_salt/py2/salt/grains/core.py", line 2480, in _linux_iqn
            final_iqn = iqn[1].rstrip()
        IndexError: list index out of range
    stdout:
```

### Tests written?

Yes

Commit:
Bugfix: iscsi initiator crash on grains fetch

Commit:
Add unit test for core._linux_iqn utility function

Commit:
Use append instead of extend

rallytime commented:
@isbm Can you fix the lint error here?

garethgreenaway commented:
This is the 3rd PR to fix this issue.  There is one by me 25 days ago, one by @terminalmage a couple days ago and now this one.

isbm commented:
@garethgreenaway mine at least has unit tests and takes care in case there is a space around `=`.

Commit:
Lintfix: C0326

isbm commented:
@rallytime I've updated the lint and rebased the branch. Please consider that my version of this fix addreses the exact problem as well as contains a unit test. In case this is not needed, please just close the PR.

rallytime commented:
Hi @isbm The PR from Gareth also has some really nice unit tests. That PR also went through quite a bit of review from other community members. I'd like to close this in favor of @garethgreenaway's PR.

For your question about handling the space around `=`, I spoke with @terminalmage today and he mentioned that there shouldn't be a space there according to [their docs](https://library.netapp.com/ecmdocs/ECMP1654943/html/GUID-14CE9411-179D-4FD9-98A9-4663841BDFD9.html).

Apologies for the extra work here. We'll try to get #43783 in ASAP so we can move forward with fixing the grain problem.

isbm commented:
@rallytime No problem! @garethgreenaway did a great job and his tests are basically the same thing I did. Thanks!

rallytime commented:
@isbm Great! Thank you for understanding. :)
